internal/tools/fetch_logs.go

```diff
@@ -0,0 +1,75 @@
+package tools
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/mark3labs/mcp-go/server"
+	"github.com/tektoncd/mcp-server/internal/params"
+	taskruninformer "github.com/tektoncd/pipeline/pkg/client/injection/informers/pipeline/v1/taskrun"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+)
+
+func fetchTaskRunLogs() server.ServerTool {
```

Comment:
@jbpratt thank you for the PR. Could you also write some unit tests for fetch_logs ?

Reply:
Hey @waveywaves , I was holding off in implementing unit tests for the tools until we establish testing with the fake informers which I think will require a refactor :) let me know if you have any ideas otherwise (I don't want to duplicate the handler into a test)

Reply:
I'll go ahead and refactor the pod fetching stuff and write a test with it!

Reply:
yess, any unit tests will work if we have to refactor for testing with informers could you create an issue for it. Also, this PR doesn't have a linked issue. If you could create one that would be great.